src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -53,6 +53,7 @@ public override async Task<string> Execute(OrchestrationContext innerContext, st
                 throw new InvalidOperationException($"The {nameof(this.FunctionInvocationCallback)} has not been assigned!");
             }
 
+            innerContext.DataConverter = this.dataConverter;

```

Comment:
I understand that our `MessagePayloadDataConverter` uses the message serialization settings, but I think I would rather be explicit and use `this.dataConverter.MessageConverter`. That way, when we add better support for error serialization in DurableTask.Core, our logic is a bit more clear.